sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationSettingsSnapshot.cs

```diff
@@ -58,15 +58,36 @@ internal ConfigurationSettingsSnapshot(string name, SnapshotStatus? status, int?
         /// <summary> Provides additional information about the status of the snapshot. The status code values are modeled after HTTP status codes. </summary>
         public int? StatusCode { get; }
         /// <summary> A list of filters used to filter the key-values included in the snapshot. </summary>
-        public IList<ConfigurationSettingFilter> Filters { get; }
+        public IList<SnapshotSettingFilter> Filters { get; }
         /// <summary> The composition type describes how the key-values within the snapshot are composed. The &apos;all&apos; composition type includes all key-values. The &apos;group_by_key&apos; composition type ensures there are no two key-values containing the same key. </summary>
         public CompositionType? CompositionType { get; set; }
         /// <summary> The time that the snapshot was created. </summary>
         public DateTimeOffset? Created { get; }
         /// <summary> The time that the snapshot will expire. </summary>
         public DateTimeOffset? Expires { get; }
+        private long? _retentionPeriod;
         /// <summary> The amount of time, in seconds, that a snapshot will remain in the archived state before expiring. This property is only writable during the creation of a snapshot. If not specified, the default lifetime of key-value revisions will be used. </summary>
-        public long? RetentionPeriod { get; set; }
+        [CodeGenMember("RetentionPeriod")]
```

Comment:
We are not generating models, so why do we need `[CodeGenMember("RetentionPeriod")]`?